packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts

```diff
@@ -65,16 +70,91 @@ type GetInitialStylesOptionsWithoutTags = Omit<GetInitialStylesOptions, 'format'
         fontStyle: 'normal',
       },
 
-      'a, button, input, select, textarea': {
-        outline: '1px solid transparent',
-        outlineOffset: '1px',
+      'a, button': {
+        position: 'relative',
+        outline: 0, // hide default focus outline
+        '&:focus::before': {
+          content: '""',
+          position: 'absolute',
+          borderRadius: borderRadiusMedium,
+          border: `${borderWidthBase} solid ${themeLight.state.focus}`,
+          ...getInsetJssStyle(-4),
+        },
+        '&:focus:not(:focus-visible)::before': {
+          border: 0,
+        },
+        '&:disabled': {
+          '&:focus::before': {
+            border: 0,
+          },
+        },
+      },
+
+      'button:focus::before': {
+        ...getInsetJssStyle(-6),
+      },
+
+      '[data-theme=dark] a, [data-theme=dark] button': {
+        '&:focus::before': {
+          border: `${borderWidthBase} solid ${themeDark.state.focus}`,
+        },
+        '&:disabled': {
+          '&:focus::before': {
+            border: 0,
+          },
+        },
+      },
+
+      // Pseudo-elements are not supported on these elements
+      'input, select, textarea': {
```

Comment:
I'd argue we shouldn't have global styles like this. What's the use case? People now wanting to use our wrapper components?